packages/react/src/toggle-button-group/root/useToggleButtonGroupRoot.ts

```diff
@@ -0,0 +1,115 @@
+    disabled = false,
+    onValueChange,
+    toggleMultiple = false,
+  } = parameters;
+
+  const [groupValue, setValueState] = useControlled({
+    controlled: value,
+    default: defaultValue,
+    name: 'ToggleButtonGroup',
+    state: 'value',
+  });
+
+  const setGroupValue = useEventCallback((newValue: string, nextPressed: boolean, event: Event) => {
+    let newGroupValue: string[] | undefined;
+    if (toggleMultiple) {
+      newGroupValue = groupValue.slice();
+      if (nextPressed) {
+        newGroupValue.push(newValue);
+      } else {
+        newGroupValue.splice(groupValue.indexOf(newValue), 1);
+      }
+    } else {
+      newGroupValue = nextPressed ? [newValue] : [];
+    }
+    if (Array.isArray(newGroupValue)) {
+      setValueState(newGroupValue);
+      onValueChange?.(newGroupValue, event);
+    }
+  });
+
+  const getRootProps = React.useCallback(
+    (externalProps = {}) =>
+      mergeReactProps<'div'>(externalProps, {
+        role: 'group',
+      }),
+    [],
+  );
+
+  return React.useMemo(
+    () => ({
+      getRootProps,
+      disabled,
+      setGroupValue,
+      value: groupValue,
+    }),
+    [getRootProps, disabled, groupValue, setGroupValue],
+  );
+}
+
+export namespace UseToggleButtonGroupRoot {
+  export interface Parameters {
+    /**
+     * The open state of the ToggleButtonGroup represented by an array of
+     * the values of all pressed `<ToggleButtonGroup.Item/>`s
+     * This is the controlled counterpart of `defaultValue`.
+     */
+    value?: readonly string[];
+    /**
+     * The open state of the ToggleButtonGroup represented by an array of
+     * the values of all pressed `<ToggleButtonGroup.Item/>`s
+     * This is the uncontrolled counterpart of `value`.
+     */
+    defaultValue?: readonly string[];
+    /**
+     * Callback fired when the pressed states of the ToggleButtonGroup changes.
+     *
+     * @param {string[]} groupValue An array of the `value`s of all the pressed items.
+     * @param {Event} event The event source of the callback.
+     */
+    onValueChange?: (groupValue: string[], event: Event) => void;
+    /**
+     * When `true` the component is disabled
+     * @false
+     */
+    disabled?: boolean;
+    /**
+     * When `false` only one item in the group can be pressed. If any item in
+     * the group becomes pressed, the others will become unpressed.
+     * When `true` multiple items can be pressed.
+     * @default false
+     */
+    toggleMultiple?: boolean;
+  }
+
+  export interface ReturnValue {
+    getRootProps: (externalProps?: GenericHTMLProps) => GenericHTMLProps;
+    /**
+     * When `true` the component is disabled
+     * @false
+     */
+    disabled: boolean;
+    /**
+     *
+     */
+    setGroupValue: (newValue: string, nextPressed: boolean, event: Event) => void;
+    /**
+     * The value of the ToggleButtonGroup represented by an array of values
+     * of the items that are pressed
+     */
+    value: readonly string[];
```

Comment:
For the value, I think it might be easier to remember (or harder to do something wrong) if it was always an array, if not we'd have to do a warning or something if you get the type wrong here

What do you think @atomiks @vladmoroz ?

Reply:
> It might be better DX if the type of the value was an array only when toggleMultiple is set.

Radix does this (different `value` type based on whether the accordion accepts multiple items) and every single time I've created a wrapped accordion component for a design system it's been painful. You can't just pass props around without getting into some weird dance trying to please TypeScript

I said this about `asChild` and will maintain throughout that we should design our API without conditional types. They make sense on the surface but harm the DX more than they help you.

So yeah I really like always having an array as a value.

Reply:
We discussed this and decided to avoid conditional types in general

Reply:
> For parity with RadioGroup, we should allow any type of value.

Updated ~ https://github.com/mui/base-ui/pull/763/commits/98746a30c22f3f4dfb6f899b2d34b148a2fa849a